### PR TITLE
Democracy 2nd iteration

### PR DESCRIPTION
**Pull request by pifragile**
Implements pallet side tasks of #347

### CODE REVIEW

File: democracy/src/lib.rs [hunk 1]
```diff
@@ -114,22 +156,11 @@ pub mod pallet {
...
-	pub(super) type VoteEntries<T: Config> = StorageDoubleMap<
-		_,
-		Blake2_128Concat,
-		ProposalIdType,
-		Blake2_128Concat,
-		VoteEntry<T::AccountId>,
-		(),
-		ValueQuery,
-	>;
 	// TODO set default value
```
Comment (brenzi) [inline democracy/src/lib.rs:159]: is this still a TODO? and does the follwoing need to be ValueQuery? Whenever possible I prefer OptionQuery

File: democracy/src/benchmarking.rs [hunk 2]
```diff
@@ -7,18 +7,22 @@ use encointer_primitives::{
...
 
-fn advance_n_blocks<T: Config>(n: u64) {
-	for _ in 0..n {
-		frame_system::Pallet::<T>::set_block_number(
-			frame_system::Pallet::<T>::block_number() + 1u32.into(),
-		);
-		frame_system::Pallet::<T>::on_initialize(frame_system::Pallet::<T>::block_number());
-	}
+fn advance_timestamp_equivalent_to_n_blocks<T: Config>(n: u64) {
+	let offset: T::Moment = (n * 6000u64)
```
Comment (brenzi) [inline democracy/src/benchmarking.rs:20]: we shouldn't hardcode block times

Comment (brenzi) [reply democracy/src/benchmarking.rs:20]: add it as a trait constant to the pallet?

Comment (brenzi) [reply democracy/src/benchmarking.rs:20]: hmm, well, if its only benchmarking IDK

**Review by brenzi — CHANGES_REQUESTED** [review]
only minor remarks. LGTM on a superficial level. before putting this in production, we'll need to review again I suggest

File: democracy/src/lib.rs (hunk 1 above)
Comment (pifragile) [reply democracy/src/lib.rs]: ok option query now

**Review by brenzi — APPROVED** [review]

File: democracy/src/lib.rs [hunk 3]
```diff
@@ -495,14 +497,14 @@ pub mod pallet {
...
-					<encointer_communities::Pallet<T>>::do_update_nominal_income(
+					<pallet_encointer_communities::Pallet<T>>::do_update_nominal_income(
 						cid,
 						nominal_income,
 					)?;
 				},
 
 				ProposalAction::SetInactivityTimeout(inactivity_timeout) => {
-					<encointer_ceremonies::Pallet<T>>::do_set_inactivity_timeout(
+					<pallet_. encointer_ceremonies::Pallet<T>>::do_set_inactivity_timeout(
```
Comment (clangenb) [inline democracy/src/lib.rs]: dafuq, how could this be merged? @brenzi, @pifragile ?

Comment (pifragile) [reply democracy/src/lib.rs]: you somehow commented on an outdated version of the diff.
image

see here:
https://github.com/encointer/pallets/pull/362/files#diff-52647a2848a1f05fd742a15c942a8f46773d88eaae8c8fee81f3cc90ad99f90dR517

Comment (clangenb) [reply democracy/src/lib.rs]: Ah, ok weird, don't know how I ended up there. XD

**Pull request merged**